fulp_modules/features/antagonists/bloodsuckers/code/antagonists/bloodsucker_datum.dm

```diff
@@ -718,6 +725,28 @@
 
 	return fullname
 
+///Different Bloodsuckers are assigned different, unique colors for their and their vassal's HUD icons.
+/datum/antagonist/bloodsucker/proc/SelectColor()
+	
+	var/list/PossibleColors = list("red", "orange", "yellow", "green", "blue", "purple", "grey", "white")
+	// Trim for uniqueness
+	for(var/datum/mind/mind_members in clan.members)
+		for(var/datum/antagonist/bloodsucker/individual_bloodsuckers in mind_members.antag_datums)
+			if(mind_members.has_antag_datum(/datum/antagonist/vassal)) // Favorite vassals do not have individual colors
+				continue
+			PossibleColors.Remove(individual_bloodsuckers.vassal_color)
+		
+		if(PossibleColors.len == 0)
+			PossibleColors = initial(PossibleColors)
+	
+	vassal_color = pick(PossibleColors)
+	update_vassal_hud(owner.current)
```

Comment:
Whys it called vassal hud if it's for /datum/antagonist/bloodsucker

Reply:
HUD depicting and designating vassals- different from things which depict things like daylight and blood count, which are also referred to as a HUD.
update_hud is afaik also already a thing.

Reply:
depicts vassals.
bloodsucker hud is already a thing, the blood and Sol overlay, and afaik `update_hud` is also something, that I don't want to overwrite.